components/msg/constellation_msg.rs

```diff
@@ -136,68 +163,17 @@ impl PipelineNamespace {
         NonZeroU32::new(self.index).expect("pipeline id index wrapped!")
     }
 
-    fn next_pipeline_id(&mut self) -> PipelineId {
-        PipelineId {
-            namespace_id: self.id,
-            index: PipelineIndex(self.next_index()),
-        }
-    }
-
-    fn next_browsing_context_id(&mut self) -> BrowsingContextId {
-        BrowsingContextId {
-            namespace_id: self.id,
-            index: BrowsingContextIndex(self.next_index()),
-        }
-    }
-
-    fn next_history_state_id(&mut self) -> HistoryStateId {
-        HistoryStateId {
-            namespace_id: self.id,
-            index: HistoryStateIndex(self.next_index()),
-        }
-    }
-
-    fn next_message_port_id(&mut self) -> MessagePortId {
-        MessagePortId {
-            namespace_id: self.id,
-            index: MessagePortIndex(self.next_index()),
-        }
-    }
-
-    fn next_message_port_router_id(&mut self) -> MessagePortRouterId {
-        MessagePortRouterId {
-            namespace_id: self.id,
-            index: MessagePortRouterIndex(self.next_index()),
-        }
-    }
-
-    fn next_broadcast_channel_router_id(&mut self) -> BroadcastChannelRouterId {
-        BroadcastChannelRouterId {
-            namespace_id: self.id,
-            index: BroadcastChannelRouterIndex(self.next_index()),
-        }
-    }
-
-    fn next_service_worker_id(&mut self) -> ServiceWorkerId {
-        ServiceWorkerId {
-            namespace_id: self.id,
-            index: ServiceWorkerIndex(self.next_index()),
-        }
-    }
-
-    fn next_service_worker_registration_id(&mut self) -> ServiceWorkerRegistrationId {
-        ServiceWorkerRegistrationId {
-            namespace_id: self.id,
-            index: ServiceWorkerRegistrationIndex(self.next_index()),
-        }
-    }
-
-    fn next_blob_id(&mut self) -> BlobId {
-        BlobId {
-            namespace_id: self.id,
-            index: BlobIndex(self.next_index()),
-        }
-    }
+    create_pipeline_namespace_function! {next_pipeline_id, PipelineId, self, PipelineIndex}
```

Comment:
nit: how about we name it a bit shorter, like `namespace_id_method`?

Reply:
renamed it